### PR TITLE
GTM billing api for getting reservation Transactions by invoice section.

### PR DESCRIPTION
**Pull request by msft-adrianma**
GTM billing api for getting reservations and transactions by invoice section.

----------------

If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/app/branch/msft-adrianma/azure-rest-api-specs/adrianma-dev-billing-Microsoft.Billing-2018-11-01-preview).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by wilcobmsft — APPROVED** [review]

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5587 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5587/commits/7bdc27f30bbacb7b776eb743ab53f0fe5ef142e3

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json [hunk 1]
```diff
@@ -1684,6 +1684,72 @@
         }
       }
     },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/invoiceSections/{invoiceSectionName}/transactions": {
```
Comment (majastrz) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: >transactions [](start = 108, length = 12)

I noticed that this doesn't have a corresponding GET for a single transaction. I'm also noticing the same pattern on all the other ones that was already pre-approved, so I can't really block based on that.

However, this is a violation of the Resource Provider Contract. Can we get that fixed?

**Review by majastrz — APPROVED** [review]
I left a comment that should get addressed later. Everything else looks good to me.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json [hunk 2]
```diff
@@ -1684,6 +1684,72 @@
...
+      "get": {
+        "tags": [
+          "Transactions"
+        ],
+        "operationId": "TransactionsByInvoiceSection",
```
Comment (veronicagg) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: suggestion: "TransactionsByInvoiceSection_List"

**Review by veronicagg — CHANGES_REQUESTED** [review]
@lmazuel I see python and other langs failing to generate SDKs based on the comments above, since it comes from the swagger to sdk script, do you mind taking a look? Thanks! I'm not sure why it's complaining about the key.

**Comment by veronicagg** [comment]
@msft-adrianma it looks like generation of the SDKs did not work well in CI above due to "{" being in the name of the PR and commit (sorry! an issue on our side). 
Will you be addressing the operation name suggestion that I mentioned above? The reason I'm suggesting NOUN_verb is that the string before the underscore turns into the name of the class and all operations under the same "NOUN" get into the same operation group. 
If you make an additional comment, CI should get re-triggered and we can see how SDK generation goes, let's not include "{" in the commit message for now. Thanks!

**Comment by msft-adrianma** [comment]
updated the PR description.

**Comment by veronicagg** [comment]
@msft-adrianma will you be making more updates or taking on the suggestion regarding the operation name?

**Comment by veronicagg** [comment]
Please review model validation errors: https://portal.azure-devex-tools.com/app/branch/msft-adrianma/azure-rest-api-specs/adrianma-dev-billing-Microsoft.Billing-2018-11-01-preview...master/static

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json [hunk 3]
```diff
@@ -421,19 +418,6 @@
               "$ref": "#/definitions/InvoiceSectionListResult"
             }
           },
-          "202": {
```
Comment (veronicagg) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: is this not a response coming from server? what's the reason for removing?

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/examples/AcceptTransfer.json [hunk 4]
```diff
@@ -21,21 +21,13 @@
...
-            "StandardAzureSubscription",
-            "AzureReservation"
-          ],
+          "creationTime": "2018-10-10T17:32:28Z",
+          "expirationTime": "2018-11-05T17:32:28Z",
           "transferStatus": "InProgress",
           "recipientEmailId": "user@contoso.com",
           "initiatorEmailId": "user@contoso.com",
-          "lastModifiedTime": "10/10/2018 9:45:01 PM",
+          "lastModifiedTime": "2018-10-10T17:32:28Z",
```
Comment (veronicagg) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/examples/AcceptTransfer.json]: Please make sure any changes made in examples, are reflecting what the service actually receives and responds with. These will feed documentation for customers.

**Review by veronicagg — CHANGES_REQUESTED** [review]
couple of comments/questions inline.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json (hunk 3 above)
Comment (msft-adrianma) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: correct. this was a bug. It is already fixed in master.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/examples/AcceptTransfer.json (hunk 4 above)
Comment (msft-adrianma) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/examples/AcceptTransfer.json]: yes. We will be doing a more detailed scrubbing of the examples. This is just a first stab at cleaning it.

**Comment by veronicagg** [comment]
@msft-adrianma looks good, would you take a look at this remaining one https://travis-ci.org/Azure/azure-rest-api-specs/jobs/514310657#L600 ?

**Review by veronicagg — APPROVED** [review]

**Comment by veronicagg** [comment]
@msft-adrianma are the APIs deployed? if so, I can merge the PR.

**Comment by msft-adrianma** [comment]
yes the apis are deployed.

**Pull request merged**